src/lib/tuya.ts

```diff
@@ -855,6 +855,64 @@ export const valueConverter = {
             return {preset: presetLookup[v], system_mode: systemModeLookup[v]};
         },
     },
+    ZWT07_schedule: {
```

Comment:
This looks the same as the already existing `ZWT198_schedule`

Reply:
ZWT198_schedule use hard-codeid dpId (`const dpId = 109;`) different and incompatible from ZWT07 (67).
I didn't want to introduce any side effects to the ZWT198 devices by modifying this converter.

Reply:
Then I propose to move the logic into a function and add the dpid as an argument

Reply:
Logic in the new converter is slightly different and doesn't use `dpId` at all (acts like simple valueconverter) and is more generic.
I think `ZWT07_schedule` can replace `ZWT198_schedule` as is (but I have nothing to check) and is a better way than extracting the function

Reply:
will address in other comments.